File: android/app/src/main/java/com/snapsort/app/home/HomeModels.kt
package com.snapsort.app.home

enum class AgentStatusKind {
    ACTIVE,
    PROCESSING,
    OFFLINE,
    INACTIVE,
}

data class AgentStatus(
    val kind: AgentStatusKind,
    val primaryText: String,
    val secondaryText: String,
)

enum class ActivityStatus {
    PROCESSING,
    EXECUTED,
    NEEDS_ATTENTION,
    UNDONE,
    DISMISSED,
    SKIPPED,
    FAILED,
    RETRYING,
}

data class ToolDestination(
    val label: String,
    val symbol: String,
)

data class ActivityItem(
    val id: String,
    val title: String,
    val status: ActivityStatus,
    val destination: ToolDestination? = null,
    val summary: String,
    val eventTime: String? = null,
    val activityTimestamp: String? = null,
    val canUndo: Boolean = false,
)

data class HomeUiState(
    val agentStatus: AgentStatus,
    val processingItem: ActivityItem? = null,
    val needsAttention: List<ActivityItem> = emptyList(),
    val recentActivity: List<ActivityItem> = emptyList(),
)

File: ios/Snapsort/Models/HomeModels.swift
import Foundation

// Presentation models for Home. Port of android/.../home/HomeModels.kt: views receive
// already-decided state and never look at confidence or re-run the policy gate.

enum AgentStatusKind: Equatable {
    case active, processing, offline, inactive
}

struct AgentStatus: Equatable {
    var kind: AgentStatusKind
    var primaryText: String
    var secondaryText: String
}

enum ActivityStatus: String, Codable, CaseIterable {
    case processing = "PROCESSING"
    case executed = "EXECUTED"
    case needsAttention = "NEEDS_ATTENTION"
    case undone = "UNDONE"
    case dismissed = "DISMISSED"
    case skipped = "SKIPPED"
    case failed = "FAILED"
    case retrying = "RETRYING"

    init(from decoder: Decoder) throws {
        let raw = try decoder.singleValueContainer().decode(String.self)
        self = ActivityStatus(rawValue: raw) ?? .executed
    }
}

struct ToolDestination: Equatable {
    var label: String
    var symbol: String
}

struct ActivityItem: Identifiable, Equatable {
    var id: String
    var title: String
    var status: ActivityStatus
    var destination: ToolDestination? = nil
    var summary: String
    var eventTime: String? = nil
    var activityTimestamp: String? = nil
    var canUndo: Bool = false
}

struct HomeUiState: Equatable {
    var agentStatus: AgentStatus
    var processingItem: ActivityItem? = nil
    var needsAttention: [ActivityItem] = []
    var recentActivity: [ActivityItem] = []
}
